src/core/common/instance.hpp

```diff
@@ -323,9 +323,10 @@ class Instance : public otInstance
     // Notifier, Settings, and MessagePool are initialized  before
     // other member variables since other classes/objects from their
     // constructor may use them.
-    Notifier    mNotifier;
-    Settings    mSettings;
-    MessagePool mMessagePool;
+    Notifier       mNotifier;
+    Settings       mSettings;
+    SettingsDriver mSettingsDriver;
```

Comment:
[commit 2]: Can we consider making `mSettingsDriver` be a member variable of `Settings` class instead?
I think `Settings` would be the only user of the driver. We can then consider not providing `Instance::Get<SettingsDriver>`.

Thoughts?

Reply:
I'm not sure this would work without additional changes. This would result in multiple instances of `mSettingsDriver` given that `ChildInfoIterator` inherits from `SettingsBase`. Correct?

Reply:
You are right. That won't work.